docs/DeveloperGuide.md

```diff
@@ -892,99 +892,12 @@ sorted order.
-
-<box type="info" seamless>
-
-**Note:** If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone AddressBook states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
-
-</box>
-
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
-
-<puml src="diagrams/UndoRedoState4.puml" alt="UndoRedoState4" />
-
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. Reason: It no longer makes sense to redo the `add n/David …​` command. This is the behavior that most modern desktop applications follow.
-
-<puml src="diagrams/UndoRedoState5.puml" alt="UndoRedoState5" />
-
-The following activity diagram summarizes what happens when a user executes a new command:
-
-<puml src="diagrams/CommitActivityDiagram.puml" width="250" />
-
-#### Design considerations:
-
-**Aspect: How undo & redo executes:**
-
-* **Alternative 1 (current choice):** Saves the entire address book.
-  * Pros: Easy to implement.
-  * Cons: May have performance issues in terms of memory usage.
-
-* **Alternative 2:** Individual command knows how to undo/redo by
-  itself.
-  * Pros: Will use less memory (e.g. for `delete`, just save the patient being deleted).
-  * Cons: We must ensure that the implementation of each individual command are correct.
-
-_{more aspects and alternatives to be added}_
+Presently, the absence of an undo and redo feature leaves room for users to inadvertently include typos in their command messages, potentially leading to unintended commands being executed. This could result in various issues, from unintentionally deleting vital data alongside the contact to inconveniencing the users themselves. Therefore, **we are planning to introduce an undo and redo feature in the future to enhance the user experience for our users.
 
+### 4.2 Edit tags feature
+Currently, the process of modifying patient tags in the PatientSync application can be cumbersome, requiring users to delete and re-add tags individually. This can lead to potential errors, especially when handling a large number of tags or making multiple changes. To address this, we are planning to introduce the EditTagsCommand feature, providing users with a more flexible and efficient way to manage patient tags. The planned EditTagsCommand feature is designed to enhance user productivity, reduce the likelihood of errors, and improve overall usability within PatientSync.
```

Comment:
if applicable(?) , mostly planned enhancements are just to let testers know about the existence of something that is planned so that they would not report a bug unnecesasrily

Reply:
it should be applicable to this scenario though right? Since Edit Tags will be a new command

Reply:
It seems that usually sample inputs and outputs are for pre-existing commands that require changes. 
If new commands are to have sample UI input and output, it would be like a design spec and would be too long.
I have also checked samples from previous semesters and most new commands are just written like this.

Have also change the undo/redo feature to another planned enhancement which has samples